docassemble/AssemblyLine/sessions.py

```diff
@@ -0,0 +1,494 @@
+    )
+
+
+def get_interview_metadata(
+    filename: str, session_id: int, metadata_key_name: str = "metadata"
+) -> Dict:
+    """Retrieve the unencrypted metadata associated with an interview.
+    We implement this with the docassemble jsonstorage table and a dedicated `tag` which defaults to `metadata`.
+    """
+    conn = variables_snapshot_connection()
+    with conn.cursor() as cur:
+        query = "select data from jsonstorage where filename=%(filename)s and tags=%(tags)s and key=%(session_id)s"
+        cur.execute(
+            query,
+            {"filename": filename, "tags": metadata_key_name, "session_id": session_id},
+        )
+        val = cur.fetchone()
+    conn.close()
+    if val and len(val):
+        return val[0]  # cur.fetchone() returns a tuple
+    return val or {}
+
+
+def get_saved_interview_list(
+    filename: str = al_session_store_default_filename,
+    user_id: int = None,
+    metadata_key_name: str = "metadata",
+) -> Tuple[Dict, int]:
+    """Get a list of saved sessions for the specified filename. If the save_interview_answers function was used
+    to add metadata, the result list will include columns containing the metadata.
+    If the user is a developer or administrator, setting user_id = None will list all interviews on the server. Otherwise,
+    the user is limited to their own sessions.
+    """
+    get_sessions_query = text(
+        """
+           SELECT  userdict.indexno
+           ,userdict.filename as filename
+           ,num_keys
+           ,userdictkeys.user_id as user_id
+           ,userdict.modtime as modtime
+           ,userdict.key as key
+           ,jsonstorage.data->'title' as title
+           ,jsonstorage.data->'description' as description
+           ,jsonstorage.data->'steps' as steps
+           ,jsonstorage.data->'original_interview_filename' as original_interview_filename
+           ,jsonstorage.data->'answer_count' as answer_count
+           ,jsonstorage.data as data
+    FROM userdict 
+    NATURAL JOIN 
+    (
+      SELECT  key
+             ,MAX(modtime) AS modtime
+             ,COUNT(key)   AS num_keys
+      FROM userdict
+      GROUP BY  key
+    ) mostrecent
+    LEFT JOIN userdictkeys
+    ON userdictkeys.key = userdict.key
+    LEFT JOIN jsonstorage
+    ON userdict.key = jsonstorage.key AND (jsonstorage.tags = :metadata)
+    WHERE (userdictkeys.user_id = :user_id or :user_id is null)
+    
+    AND
+    (userdict.filename = :filename OR :filename is null)
+    ORDER BY modtime desc 
+    LIMIT 500;
+    """
+    )
+
+    # TODO: decide if we need to handle paging
+
+    if not filename:
+        filename = None  # Explicitly treat empty string as equivalent to None
+    if user_id is None:
+        user_id = user_info().id
+
+    if user_id == "all":
+        if user_has_privilege(["developer", "admin"]):
+            user_id = None
+        else:
+            user_id = user_info().id
```

Comment:
That's a good catch--just curious though, how did you trigger that? Were you manually running the `interview_list_html()` method in a new YAML file?

Reply:
Yep, I ran it with `user_id == "all"` just to see what it would do, both while logged in and while not logged in. I think it was trigged when I wasn't logged in.